pkgs/tools/networking/smartdns/default.nix

```diff
@@ -0,0 +1,55 @@
+
+{ stdenv, fetchFromGitHub, openssl, fetchpatch }:
+
+stdenv.mkDerivation rec {
+  pname = "smartdns";
+  version = "30";
+  src = fetchFromGitHub {
+    owner = "pymumu";
+    repo = pname;
+    rev = "Release${version}";
+    sha256 = "1zj1ajvdnpmdal87j7fyn0xdd0ydjii1nvgwgacv2j311b6fdrk7";
+  };
+
+  buildInputs = [ openssl ];
+
+  # TODO: A patch is needed by now since the upstream doesn't follow the FHS guideline. It should not be the case in the future.
+  patches = [
+    (fetchpatch {
+      url =
+        "https://git.archlinux.org/svntogit/community.git/plain/trunk/systemd.patch?h=packages/smartdns&id=c15e81742c93e24c47de59a956f1d0d0f7156f0a";
+      name = "systemd.nix";
+      sha256 = "0jxzsgxz6y1a4mvwk6d5carxv4nxsswpfs1a6jkplzy2gr0g8281";
+    })
+  ];
+
+  patchFlags = [ "-p1" "systemd/smartdns.service" ];
```

Comment:
This should not be necessary any more.

Reply:
No, it is required since the patch is in a way in which we need to pipe in `orginfile`

Reply:
No, it is required since the patch is in a way in which we need to pipe in `orginfile`

Reply:
Hmm, you are right, the Arch patch show wrong path. I suggest passing `extraPrefix = "systemd";` to `fetchpatch` to mitigate the need for `patchFlags`. The hash will need to be changed if you do that since it changes the patch.

Reply:
No, the `.old` and `.new` convention seems still prevent me from getting there. I have tried `extraPrefix = "systemd";` and `extraPrefix = "systemd/";`